elide-contrib/elide-swagger/src/test/java/com/yahoo/elide/contrib/swagger/JsonApiModelResolverTest.java

```diff
@@ -74,11 +74,86 @@ public void testModelResolution() throws Exception {
         Assert.assertTrue(relationships.getProperties().containsKey("books"));
         Assert.assertTrue(relationships.getProperties().containsKey("exclusiveAuthors"));
     }
+
+    @Test
+    public void testDescription() {
+        ModelConverters converters = ModelConverters.getInstance();
+        converters.addConverter(new JsonApiModelResolver(dictionary));
+
+        Map<String, Model> models = converters.readAll(Publisher.class);
+
+        Resource publisherModel = (Resource) models.get("publisher");
+
+        ObjectProperty attributes = (ObjectProperty) publisherModel.getProperties().get("attributes");
+
+        String phoneDescription = attributes.getProperties().get("phone").getDescription();
+        Assert.assertEquals(phoneDescription,  "Phone number");
+    }
+
+    @Test
+    public void testExample() {
+        ModelConverters converters = ModelConverters.getInstance();
+        converters.addConverter(new JsonApiModelResolver(dictionary));
+
+        Map<String, Model> models = converters.readAll(Publisher.class);
+
+        Resource publisherModel = (Resource) models.get("publisher");
+
+        ObjectProperty attributes = (ObjectProperty) publisherModel.getProperties().get("attributes");
+
+        Object phoneExample = attributes.getProperties().get("phone").getExample();
+        Assert.assertEquals(phoneExample,  "555-000-1111");
+    }
+
+    @Test
+    public void testConcatenatedDescriptionAndPermissions() {
+        ModelConverters converters = ModelConverters.getInstance();
+        converters.addConverter(new JsonApiModelResolver(dictionary));
+
+        Map<String, Model> models = converters.readAll(Book.class);
+
+        Resource bookModel = (Resource) models.get("book");
+
+        ObjectProperty attributes = (ObjectProperty) bookModel.getProperties().get("attributes");
+
+        String isbnDescription = attributes.getProperties().get("year").getDescription();
+        Assert.assertEquals(isbnDescription,  "Year published\nRead Permissions : (Principal is author OR Principal is publisher)");
+    }
+
+    @Test
+    public void testRequired() {
+        ModelConverters converters = ModelConverters.getInstance();
+        converters.addConverter(new JsonApiModelResolver(dictionary));
+
+        Map<String, Model> models = converters.readAll(Book.class);
+
+        Resource publisherModel = (Resource) models.get("book");
+
+        ObjectProperty attributes = (ObjectProperty) publisherModel.getProperties().get("attributes");
+
+        boolean titleRequired = attributes.getProperties().get("title").getRequired();
+        Assert.assertTrue(titleRequired);
+    }
+
+    @Test
+    public void testReadOnly() {
+        ModelConverters converters = ModelConverters.getInstance();
+        converters.addConverter(new JsonApiModelResolver(dictionary));
+
+        Map<String, Model> models = converters.readAll(Book.class);
+
+        Resource publisherModel = (Resource) models.get("book");
```

Comment:
I assume you meant bookModel.  Same comment in prior test.

Reply:
Done
Also refactored the test class as there has been duplication across tests